src/cloudflare/internal/vectorize.d.ts

```diff
@@ -126,25 +126,81 @@ interface VectorizeMatches {
 /**
  * Results of an operation that performed a mutation on a set of vectors.
  * Here, `ids` is a list of vectors that were successfully processed.
+ *
+ * @deprecated This is a legacy return type used during the Vectorize beta.
  */
-interface VectorizeVectorMutation {
+interface VectorizeVectorMutationLegacy {
   /* List of ids of vectors that were successfully processed. */
   ids: string[];
   /* Total count of the number of processed vectors. */
   count: number;
 }
 
 /**
-* Results of an operation that performed a mutation on a set of vectors
-* with the v2 version of Vectorize.
-* Here, `mutationId` is the identifier for the last mutation processed by Vectorize.
-*/
-interface VectorizeVectorMutationV2 {
-  /* The identifier for the last mutation processed by Vectorize. */
+ * Results of an operation that performed a mutation on a set of vectors with the post-beta version of Vectorize.
+ * Actual mutations are processed async where the `mutationId` is the unique identifier for the operation.
+ */
+interface VectorizeVectorMutation {
```

Comment:
We will need some more changes to support types for the new V2 operations. Furthermore, I think there would be some breaking changes we need to account for the query V2, get_by_ids V2 and delete_by_ids V2 responses. For reference: https://github.com/cloudflare/workers-sdk/blob/719cfad52a0e2ca8080eed99f0cbf29a5f8f42cd/packages/wrangler/src/vectorize/types.ts